src/metric/binary_metric.hpp

```diff
@@ -263,5 +263,126 @@ class AUCMetric: public Metric {
   std::vector<std::string> name_;
 };
 
+
+/*!
+* \brief Average Precision Metric for binary classification task.
+*/
+class AveragePrecisionMetric: public Metric {
+ public:
+  explicit AveragePrecisionMetric(const Config&) {
+  }
+
+  virtual ~AveragePrecisionMetric() {
+  }
+
+  const std::vector<std::string>& GetName() const override {
+    return name_;
+  }
+
+  double factor_to_bigger_better() const override {
+    return 1.0f;
+  }
+
+  void Init(const Metadata& metadata, data_size_t num_data) override {
+    name_.emplace_back("ap");
```

Comment:
@btrotta 
Was name left `ap` (instead of `average_precision`, https://github.com/microsoft/LightGBM/pull/3347#discussion_r482112568) intentionally?

Reply:
Oops, this was a mistake, fixed now.